Automatic Updates for Bayanat

level09 opened the pull request:
## Summary
- Implements comprehensive automatic update system for Bayanat
- Adds version management and update checking capabilities
- Integrates update functionality into the admin interface

## Key Features
- Automatic version checking against GitHub releases
- One-click update process from the admin panel
- Rollback capability for failed updates
- Comprehensive logging and error handling
- Security considerations with signature verification support

## Implementation Details
- New update module with version comparison logic
- Admin UI integration with update status display
- Background task support for update operations
- Configuration management for update settings

## Test Plan
- [ ] Test version checking against GitHub API
- [ ] Verify update process in development environment
- [ ] Test rollback functionality
- [ ] Validate error handling and logging
- [ ] Review security implications

level09 committed:
Simplify migration system by removing obsolete bootstrap code

- Remove _initialize_db_state() and mark_migrations_applied() functions
- Use convention that migration_history table exists (created via model in fresh installs)
- Organize imports with pathlib at top since used throughout
- Clean session handling with dedicated sessionmaker for migration isolation

level09 committed:
Add migration history table creation for automatic update system

- Single migration file creates migration_history table for tracking applied migrations
- Enables atomic database updates from UI without server infrastructure complexity
- Convention-based approach where this table is guaranteed to exist

level09 committed:
Remove obsolete comment

level09 committed:
Add test migration for automatic update verification

level09 committed:
Remove obsolete command registration

Remove mark_migrations_applied command registration that was deleted in migration system simplification

level09 committed:
Revert "Add test migration for automatic update verification"

This reverts commit 2eab2fd091e6f24b730e5b890d5c99ad0bc8ae2f.

level09 committed:
bump version to 2.9.25

level09 committed:
Reapply "Add test migration for automatic update verification"

This reverts commit b30b546ac6f46a4b10dfb340befc314d01b34924.

level09 committed:
Revert "Reapply "Add test migration for automatic update verification""

This reverts commit ebdaef4b391da9bd4c385a6f003b3672514b04e5.

level09 committed:
Simplify backup_db and add integrity verification

level09 committed:
Add logging for backup verification

level09 committed:
Add automatic rollback on update failure

  - Track git commit and backup file before update starts
  - On failure: git reset, restore dependencies, restore database, restart service
  - Reuse existing restart_service() function instead of duplicate curl command
  - Move imports to top of file (remove duplicate subprocess/pathlib in function)
  - Simplify error handling with unified exception handler

level09 committed:
Add logging points for system update steps

- Log backup creation, new commit after pull, dependency installation, migration completion, and service restart.
- Update final log message to indicate successful completion

level09 committed:
Add failing test migration for rollback verification

level09 committed:
bump version to 2.9.26

level09 committed:
Fix test migration to fail immediately instead of hanging

level09 committed:
bump version to 2.9.27

level09 committed:
Fix backup function to work both as CLI command and internal call

level09 committed:
bump version to 2.9.28

level09 committed:
bump version to 2.9.29

level09 committed:
Refactor backup/restore to work as both CLI and internal functions

level09 committed:
bump version to 2.9.30

level09 committed:
Fix database restore blocking during automatic update rollback

  - Terminate active PostgreSQL connections before pg_restore
  - Prevents pg_restore --clean from hanging on active sessions
  - Gracefully handles permission errors with suppress()

level09 committed:
bump version to 2.9.31

level09 committed:
Remove failing test migration and add dummy update test migration

level09 committed:
bump version to 2.9.32

level09 committed:
Remove set_version command; version is now automatically updated during system updates

level09 committed:
Improve automatic update system reliability and idempotency

  - Remove version check to allow idempotent updates (reruns, hotfixes,
  repairs)
  - Simplify version update logic, remove unnecessary error wrapper
  - Replace non-ASCII symbols (✓, ⚠) with plain text for compatibility

level09 committed:
bump version to 2.9.33

level09 committed:
Use database version as source of truth for app version in system status API

level09 committed:
bump version to 2.9.34

level09 committed:
Refactor restore_backup function to simplify success handling

  - Remove redundant return code checks after subprocess.run
  - Log and echo success message directly if command completes without timeout

level09 committed:
Refactor error handling in restore_backup function

  - Consolidate error logging and user feedback for database restoration failures

level09 committed:
Refactor command logging and user feedback

level09 committed:
Add database utility to centralize parsing PostgreSQL URI and refactor pg_dump command

level09 committed:
Simplify database backup and restore commands to eliminate URI parsing utility

level09 committed:
Remove potential DB password leak in logs during backup and restore

level09 committed:
bump version to 2.9.35

level09 committed:
add failing migraiton for testing

level09 committed:
bump version to 2.9.36

level09 committed:
Return status tuple from update function instead of raising exceptions

Changes update flow from exception-based to explicit return values. This eliminates duplicate error logging when rollback succeeds - the API now logs once with appropriate level (warning for rolled-back failures, info for success) instead of catching and re-logging the same exception.

level09 committed:
bump version to 2.9.37

level09 committed:
Modernize maintenance.py with Python 3 features

- Add type hints to all functions
- Replace os.path with pathlib for cleaner path operations
- Use pathlib methods (.exists(), .unlink(), .write_text(), .read_text())
- Simplify directory creation with mkdir(exist_ok=True)
- Clean up imports and exception handling
- Maintain full backward compatibility and functionality

level09 committed:
Remove test migration file that intentionally contained a syntax error for rollback verification

level09 committed:
Add periodic version check Celery task

- Check GitHub API for new releases
- Compare against pyproject.toml version
- Store in Redis when update available
- Graceful error handling with logging

level09 committed:
Register version check as periodic task

- Run every 12 hours

level09 committed:
Add lightweight version check endpoint

- Read from Redis cache instead of GitHub API
- Single Redis read per request (< 1ms)
- Admin-only endpoint for badge queries

level09 committed:
Clear update notification after successful update

- Delete Redis key after version updated in DB
- Badge disappears immediately for all admins

level09 committed:
Refactor version check logic into a standalone function and update Celery task to delegate to it

level09 committed:
Add & register 'check_updates' cli command to check for available updates and display status. using centralized version from celery tasks

level09 committed:
bump version to 2.9.38

level09 committed:
bump version to 2.9.39

level09 committed:
bump version to 2.9.40

level09 committed:
bump version to 2.9.41

level09 committed:
Remove update check functionality from dashboard template

level09 committed:
Add update check functionality to admin nav bar

- Introduced a tooltip button for admins to check for system updates.
- Implemented polling for update status every minute.
- Added methods to handle update visibility and fetch update information.
- Displayed appropriate messages based on update status (available, checking, or up to date).
- Included app version in layout for reference.

level09 committed:
bump version to 2.9.42

level09 committed:
Implement update handling and update nav bar interaction

- Added `updating` state to manage update process visibility.
- Enhanced update icon logic to reflect both updating and checking states.
- Implemented confirmation dialog for system updates, with appropriate success/error handling and user notifications.

level09 committed:
bump version to 2.9.43

level09 committed:
Update tooltip positioning in admin nav bar

- Adjusted tooltip location to "top end" for better visibility.

level09 committed:
bump version to 2.9.44

level09 committed:
Adjust & fix tooltip positioning in admin nav bar for improved visibility

level09 committed:
bump version to 2.9.45

level09 committed:
Adjust tooltip offset in admin nav bar

level09 committed:
bump version to 2.9.46

level09 committed:
bump version to 2.9.47

tarekio commented on enferno/commands.py:
```diff
@@ -322,3 +428,356 @@ def generate_config() -> None:
+        logger.info("System update completed successfully")
+        return (True, "System updated successfully")
+
+    except Exception as e:
```
very generic and catchall. We should instead raise specific exceptions above for each stage, and catch those individually and log them correctly, and then have a catch all at the end.

level09 replied:
Each stage already raises its own specific exception, so the single catch just guarantees the rollback runs every time.  

Splitting it up would mean duplicating that cleanup, or worse, missing a path, because we can’t realistically predict every failure mode. 

Also , the original exception still gets logged in full, so we can investigate the details straight from the logs

tarekio reviewed (dismissed):
Some issues, and I'm still getting stuck at setup wizard, just need to restart services after install.

level09 committed:
Fix systemctl path references for Ubuntu compatibility (Fixes system restart after installation/setup wizard completion)

Use full /usr/bin/ paths for systemctl and systemd-run to ensure sudoers rules match actual command execution. Prevents permission errors on Ubuntu systems where PATH resolution may differ from sudoers expectations.

level09 commented on enferno/admin/views.py:
```diff
@@ -6589,6 +6595,131 @@ def api_bulletin_web_import(validated_data: dict) -> Response:
+@admin.route("/api/system/status", methods=["GET"])
+@auth_required("session")
+@roles_required("Admin")
+def api_system_status() -> Response:
```
This endpoint remains from the original flow and acts as a simple “check now” ping that reaches GitHub directly and returns the latest status. 

Redis stays in sync through the Celery job, (and the CLI shortcut I added does the same).

If you prefer that a manual button refresh the shared flag too, I can switch it to call perform_version_check() so Redis updates whenever a manual check is done.

level09 committed:
Merge main branch - resolve .gitignore conflict

level09 committed:
Implement system update celery background task with status management & Redis integration

level09 committed:
add Redis-based update status management utility

level09 committed:
Remove redundant check for ongoing updates in system update task (check early on in views.py)

level09 committed:
Add notification configuration for system update availability

level09 committed:
Refactor system update task to improve status tracking and lock management

level09 committed:
Add constant for system update availability notification

level09 committed:
move update logic from sync to background tasks with status polling

level09 committed:
update cli commands to use simplified update task

level09 committed:
fix task registration order and improve version check

level09 committed:
simplify background update task with redis status tracking

level09 committed:
consolidate update locking and status into single redis key

level09 committed:
bump version to 2.9.50

level09 committed:
unify locking logic in task instead of api, simplify design

level09 committed:
switch version check to manifest instead of parsing github tags

level09 committed:
add version manifest file for update checking

level09 committed:
always update cache on version check to prevent stale dashboard data

level09 committed:
add system update page with version check and update functionality

level09 committed:
refactor update handling in nav bar and global mixin; remove direct update click logic and simplify update icon state

level09 committed:
add system update management page and API for update history retrieval

level09 committed:
add UpdateHistory model for tracking completed system updates

level09 committed:
refactor update history display; simplify icon usage and improve version formatting

level09 committed:
add update_history table migration sql

level09 committed:
fix api_get_system_update_history

level09 committed:
Update Bayanat version to 2.9.50 in uv.lock

level09 committed:
Log successful updates to UpdateHistory table

- Record version_to when update completes

level09 committed:
bump version to 2.9.51

level09 committed:
fix: update Redis cache after successful system update instead of deleting it

After a successful update, maintain accurate version cache so the UI always
shows the correct current/latest versions. Previously, deleting the cache
caused the UI to show both versions as equal even if they differed, because
the cached latest version was lost.

Changes:
- Instead of deleting bayanat:update:latest, update it with the new version
- This preserves version history in Redis for accurate dashboard display
- Timestamp updated to reflect when update actually completed
- Fixes issue where UI showed 'up to date' after updating to wrong version

level09 committed:
bump version to 2.9.52

level09 committed:
bump version to 2.9.53

level09 committed:
feat: add fresh version check and simplify page code

- Add fresh=true parameter to version check endpoint (bypasses Redis cache)
- System update page checks GitHub directly when user clicks Check button
- Extract applyVersionInfo() helper to eliminate code duplication

level09 committed:
Refactor repository URL handling in install script

level09 committed:
Improve install script reliability and security

  - Add jq for JSON parsing in API handler
  - Fix REPO variable passing to gen-env.sh
  - Align all sudoers rules with actual command usage
  - Fix uv --directory flag (was using invalid -C)
  - Add migration wrapper for clean environment handling
  - Update Caddy TLS flags and make GPG install idempotent
  - Add pipefail for uv installation safety

level09 committed:
bump version to 2.9.5٤

level09 committed:
bump version to 2.9.54

level09 committed:
whitelist update status endpoint from maintenance mode

level09 committed:
bump version to 2.9.55

level09 committed:
bump version to 2.9.56

level09 committed:
Add system update grace period configuration

level09 committed:
Add SYSTEM_UPDATE_PENDING notification status constant

level09 committed:
Add system update pending notification configuration

level09 committed:
Add new notification methods to notify all users

level09 committed:
Clear update schedule flag once performing system update

level09 committed:
Implement graceful system update scheduling with notificaiton and
validation handling

level09 committed:
Introduces a new API endpoint to schedule system updates with a grace
period. Allows administrators to schedule updates with an optional flag
to skip backup

level09 committed:
Frontend: Add system update menu with scheduled update option

apodacaduron committed:
add new system updates section on system administration page

apodacaduron committed:
tweak slug to use middle dash

apodacaduron committed:
add new system updates page

apodacaduron committed:
updates to navigation drawer

apodacaduron committed:
spacing adjustmenst to drawer

apodacaduron committed:
Merge branch 'automatic-updates' of https://github.com/sjacorg/bayanat into BYNT-1487-Implement-Auto-Update-design

apodacaduron committed:
remove system updates page and use existing system update page

apodacaduron committed:
update version on ui

level09 committed:
Enhance system update process by logging out users before maintenance mode and updating notification message to inform users of automatic logout during updates.

level09 committed:
Fix bug using wrong redis db to log out users

apodacaduron committed:
tweak font weight

apodacaduron committed:
remove expansion panels and add back release notes text

apodacaduron committed:
render table and history items in drawer

apodacaduron committed:
Merge branch 'BYNT-1487-Implement-Auto-Update-design' of https://github.com/sjacorg/bayanat into automatic-updates

apodacaduron committed:
update system update page path on routes

apodacaduron committed:
update duplicated translation

apodacaduron committed:
add missing translation to system update page

apodacaduron committed:
update date formagt

apodacaduron committed:
add translations to maintenance page

level09 committed:
Add functions to check and retrieve scheduled update status and time

level09 committed:
Refactor update status API to include scheduled update information and improve error handling

level09 committed:
Add API endpoint to fetch release notes from GitHub with Redis caching

level09 committed:
Merge branch 'automatic-updates' of github.com:sjacorg/bayanat into automatic-updates

level09 committed:
Lint fixes, formatting , cleanup

apodacaduron committed:
add back release notes button

apodacaduron committed:
fix date format typo

apodacaduron committed:
add support for markdown, render release notes, disable update button if theres an update already scheduled

apodacaduron committed:
Merge branch 'automatic-updates' of https://github.com/sjacorg/bayanat into automatic-updates

apodacaduron committed:
if user is admin and notification includes System Update redirect to updates page

apodacaduron committed:
fix issue with menu and disabled button

tarekio commented on enferno/admin/models/Notification.py:
```diff
@@ -172,6 +172,26 @@ def create_for_admins(
+        title,
+        message,
+        category=NotificationCategories.ANNOUNCEMENT.value,
+        is_urgent=False,
```
Set to true

level09 replied:
Done

apodacaduron committed:
remove system updates entry for automatic updates

level09 committed:
Enhance version check to bypass CDN cache by adding cache control headers to the request

level09 committed:
Merge branch 'automatic-updates' of github.com:sjacorg/bayanat into automatic-updates

level09 committed:
Remove unused import and unused variables in maintenance module

apodacaduron committed:
remove pop date field

level09 committed:
Enhance system update task to include user ID for tracking updates initiated by users

level09 committed:
Update UpdateHistory model to include status and user tracking

level09 committed:
update migration file for update history to include additional fields we added to the sql alchemy model

level09 committed:
Enhance system update task to log target version and update status on success or failure

level09 committed:
Update system update functions to include initiator user ID

level09 committed:
Merge branch 'automatic-updates' of github.com:sjacorg/bayanat into automatic-updates

tarekio reviewed (commented):
Just a few tiny issues to address.

Separately we should add more integrity checks on update, we can address this in a future PR.

level09 committed:
Add version_from field to UpdateHistory model for tracking version changes

level09 committed:
Update system update task to log current version (from) along with target version in UpdateHistory

level09 committed:
Add version_from column to update_history table migration sql

level09 committed:
Fix critical concurrency bug and simplify update task

      - Fix concurrency bug where concurrent update attempts could clear active
        update's locks by moving lock acquisition check before try block
      - Remove redundant exception handling around .save() calls that already
        handle exceptions internally
      - Extract session logout to reusable logout_all_users() helper in
        maintenance.py for better separation of concerns
      - Add Redis failure backup storage to survive database rollbacks
      - Improve exception handling with early version capture for better
        failure tracking

level09 committed:
Extract rollback logic to reusable function in update_utils

level09 committed:
Added missing celery restart after system upgrade

level09 committed:
Set is_urgent flag to True in Notification model for announcements

level09 committed:
Remove unnecessary pass statement in verify_backup function

level09 committed:
Fix mixed implicit/explicit returns in send_notification_to_all_users

level09 committed:
Add return statement to register_maintenance_middleware (Fix code quality)

level09 committed:
Remove unused Role import from user views.py

level09 committed:
Code quality fixes

level09 committed:
- Restrict maintenance route access to only when maintenance mode is active
    Prevent information disclosure and confusing UX when system is operational

level09 committed:
Update error handling in system update history API to address information exposure

level09 committed:
bump version to 2.9.57

level09 committed:
bump version to 2.9.58

level09 committed:
bump version to 2.9.59

level09 committed:
Add configurable version check interval to settings and update task setup integration

level09 committed:
Merge main into automatic-updates: integrate dynamic form changes with system update features

level09 committed:
Add lean db_doctor function for system update validation

- Add db_doctor() function to db_alignment_helpers with simple tuple return
- Checks critical tables (actor, bulletin, incident, user, etc.) exist
- Validates columns match between models and database schema
- Integrates into system update as step 5 (after migrations, before version update)
- Rename CLI command: flask check-db-alignment -> flask db-doctor
- Organize imports: move inline imports to module level
- Returns (healthy, diagnosis) tuple for easy integration in automated processes

The new function is lean and focused on catching critical schema issues
during system updates, while the existing DBAlignmentChecker class
remains available for detailed schema analysis.

level09 committed:
Update error handling in db_doctor_cli to raise SystemExit instead of ClickException

level09 committed:
Add dynamic field validation to db_doctor

level09 committed:
Refactor db_doctor_cli to use click.echo for output instead of logger, improving CLI feedback during database health checks.

level09 committed:
Add dummy migration for update testing

level09 committed:
bump version to 2.9.60

level09 committed:
Refactor update mechanism to use git tags instead of manifest

- Replace manifest.json with direct git tag queries for version checking
- Update system_update to fetch and checkout release tags instead of git pull
- Modify install.sh to clone latest release tag for stable installations
- Update API handler to use tag-based updates
- Add sudoers permissions for tag operations

Benefits:
- No GitHub API rate limits (git protocol has no limits)
- One less file to maintain (no manifest.json)
- Updates to stable releases only (not random commits)
- Simpler release workflow (just tag and push)
- Easy version pinning and rollback

level09 committed:
Fix sudoers line length limit by splitting long commands

Split long sudoers lines with backslash continuation to avoid
syntax errors from line length limits in sudoers files.

level09 committed:
Refactor installation script to clone repository with full history and checkout latest release tag

level09 committed:
bump version to 2.9.62

level09 committed:
Hardcode BAYANAT_REPO in settings for security

Remove environment variable reading for repository name and hardcode
it in Config to prevent potential repository injection attacks.

Changes:
- Add BAYANAT_REPO = 'sjacorg/bayanat' to Config class
- Update perform_version_check() to use cfg.BAYANAT_REPO
- Update get_release_notes() to use Config.BAYANAT_REPO
- Remove os.environ.get('BAYANAT_REPO') calls

Security improvement: Prevents attackers from manipulating the
repository source via environment variables.

level09 committed:
Remove obsolete manifest.json file

The manifest file is no longer used after refactoring to tag-based
updates. Version checking now uses git ls-remote to query tags
directly from the repository.

level09 committed:
Remove BAYANAT_REPO from env file generation

BAYANAT_REPO is now hardcoded in settings.py for security,
so no need to generate it in .env files.

level09 committed:
Fix update history showing duplicate versions

level09 committed:
Use remote tags instead of local tags for updates

apodacaduron committed:
on setup wizard when user clicks proceed to dashboard do a 10 sec countdown

apodacaduron committed:
remove space

level09 committed:
Remove duplicate v prefix in version display

level09 committed:
bump version to 2.9.63

level09 committed:
bump version to 2.9.64

level09 committed:
Simplify version management to use pyproject.toml as single source of truth

apodacaduron committed:
Revert "remove space"

This reverts commit cf22e4a570056c768d47bc493843e41c8d2f2f23.

apodacaduron committed:
Revert "on setup wizard when user clicks proceed to dashboard do a 10 sec countdown"

This reverts commit 1cce509e0928a782b63caac708c2989be6c8e7e2.

apodacaduron committed:
set 10 second delay for setup wizard

level09 committed:
Fix version comparison to use semantic versioning

level09 committed:
bump version to 2.9.65

apodacaduron committed:
tweak css markdown content

level09 committed:
Merge branch 'automatic-updates' of github.com:sjacorg/bayanat into automatic-updates

apodacaduron committed:
adjust code tag colors on markdown

apodacaduron committed:
Merge branch 'automatic-updates' of https://github.com/sjacorg/bayanat into automatic-updates

level09 committed:
bump version to 2.9.66

level09 committed:
Merge branch 'automatic-updates' of github.com:sjacorg/bayanat into automatic-updates

level09 committed:
remove dummy migration

apodacaduron committed:
enable md features like html, typographer and links

apodacaduron committed:
tweak css to support blockquotes

apodacaduron committed:
add failed message and tweak spacing

apodacaduron committed:
tweak notification redirect

apodacaduron committed:
tweak text for failed update

level09 committed:
Clear release notes cache after successful system update

level09 committed:
Merge branch 'automatic-updates' of github.com:sjacorg/bayanat into automatic-updates

level09 committed:
Fix git tag collision issue in system updates

Add --force flag to git fetch to handle tag conflicts between local and remote.
This prevents update failures when local tags point to different commits than
remote tags with the same name. Remote tags are the source of truth for releases.

level09 committed:
Don't clear maintenance mode in finally block after update

Previously maintenance was cleared immediately after restart signal,
causing a race condition where users could access the old version
during the ~15 seconds between restart signal and actual restart.

Now maintenance stays enabled until the maintenance page polls and
detects update completion, then auto-clears itself.

level09 committed:
Refactor maintenance mode check to simplify conditions

level09 committed:
auto-clear maintenance mode after successful updates on app startup

level09 committed:
Signal app to auto-clear maintenance mode on startup after updates

level09 committed:
Implement auto-clear of maintenance mode after successful rollback

level09 committed:
Add auto-refresh script to maintenance page for status updates

level09 committed:
Redirect to maintenance page after initiating system update (immediate) instead of polling for status since users are being logged out

level09 committed:
bump version to 2.9.67

tarekio committed:
ux: improve update history display and enhance system update messaging

tarekio committed:
enhance system update messaging with icons and update release notes display

tarekio committed:
Merge branch 'main' into automatic-updates

level09 committed:
Remove automatic stash restoration after system updates

 Replace auto-apply logic with manual review instruction to prevent merge conflicts

level09 committed:
Fix missing null check in update release notes display

level09 committed:
Add BACKUP_RETENTION_DAYS configuration to settings

level09 committed:
Add cleanup_old_backups function to manage old backup files

level09 committed:
Add backup cleanup logic to daily_backup_cron for managing local backups

level09 committed:
Refactor bulk update functions to use 'chunk' instead of 'group' for clarity (Fix shadow variable issue). Update conditional checks for boolean values to use 'is' for better readability and following standards.

level09 committed:
Remove logging of maintenance directory readiness in get_maintenance_file_path function avoid spamming log file

level09 committed:
Set logger propagation to False in get_logger function to prevent log messages from being passed to parent loggers and written twice

apodacaduron committed:
Merge branch 'main' of https://github.com/sjacorg/bayanat into automatic-updates

apodacaduron committed:
Merge branch 'automatic-updates' of https://github.com/sjacorg/bayanat into automatic-updates

apodacaduron committed:
add back accidentaly removed css from form builder

level09 committed:
Simplify API handler to service restarts only, reduce daemon permissions

apodacaduron committed:
refactor confirmation dialog for immediate and scheduled update

level09 committed:
Replace git stash with fail-fast check for uncommitted changes

Production environments should not have local modifications.
Instead of stashing and continuing, fail immediately with a clear error.
This simplifies the update logic and removes a potential failure point.

level09 committed:
Use remote tags in install.sh for consistency with update flow

Changed from local `git tag --list` to `git ls-remote` to fetch
latest release tag directly from remote. This matches the update
logic in commands.py and ensures the source of truth is always
the remote repository.

level09 committed:
Increase auto_clear TTL from 60s to 300s for maintenance flag

60 seconds was too tight for edge cases like slow DB connections
or heavy server load during service restart. 300 seconds provides
ample safety margin with no downside since the flag is deleted
immediately on app startup anyway.

level09 committed:
Remove unused target_version variable in update task

level09 committed:
Simplify maintenance clearing with startup check

Replace auto_clear Redis flag mechanism with simpler startup check:
- If maintenance is on but no update running or scheduled, clear it
- Removes TTL race conditions and flag-setting complexity
- Self-healing: any restart clears stale maintenance locks
- One mechanism instead of two (file + flag)

level09 committed:
Fix fail-fast check to ignore untracked files

The update check now uses --untracked-files=no to ignore runtime
directories like instance/, logs/, etc. Only modified tracked files
block updates.

level09 committed:
Use origin remote for update checks + add system_info migration

- Changed git ls-remote to use 'origin' instead of hardcoded GitHub URL
  This allows testing updates with a local repo by temporarily changing origin
- Added migration to create system_info table for storing key-value pairs

level09 committed:
Fix restart_service to not crash on API failure

- Log API failures instead of silently falling through
- In production, don't fall through to SIGHUP (won't work anyway)
- In dev mode, catch PermissionError gracefully
- Increase API timeout from 2s to 5s

apodacaduron committed:
Merge branch 'main' of https://github.com/sjacorg/bayanat into automatic-updates

level09 committed:
Merge branch 'main' into automatic-updates

# Conflicts:
#	pyproject.toml

level09 committed:
Merge remote-tracking branch 'origin/main' into automatic-updates

# Conflicts:
#	enferno/templates/dashboard.html
#	enferno/templates/layout.html
#	uv.lock

apodacaduron committed:
Merge branch 'main' of https://github.com/sjacorg/bayanat into automatic-updates

level09 commented:
Closing in favor of an upcoming full rewrite.
